### PR TITLE
2.1.1 Assignments Mastin

### PR DESCRIPTION
**Pull request by mmastin**
Base goals

### CODE REVIEW

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# making_forecasts - "Matthew Mastin"

# Part 1 - API Load
***
**What went well:**

- Great job using datetime! 

```python
df['Date'] = pd.to_datetime(df['Date'])
df = df.set_index('Date')


```

- Clean and simplisitic code with a nice graph(nice figure size as well):
```python

df.plot(figsize=(16,9))

```

-Nice use of Regex and date_range:

```python
df = df.reindex(pd.date_range(start='2015-07-01', end='2021-05-23'))
df.plot(figsize=(16,9));
```
**Where you could of improved on:**

-I'd add x and y label title :

```python
plt.ylabel('Title here')
plt.xlabel('Title here')
```


_____________________
# Graphing

***

**What went great:**

-Splicing is great :

```python

df['2019-04': '2019-05'].plot(figsize=(16,9))

```



**-Great use of forcasting :**
```python

fig2 = m.plot_components(forecast)
```




# What to work on
- [x] Practice formatting and regex
- [x] Comment more

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# evaluating_forecasts - "Matthew Mastin"

# Part 1 - Loading Data/Analysis 
***
**What went well:**

- Good job with importing in clean, reusable code: 

```python

project = 'en.wikipedia'
access = 'all-access'
agent = 'user'
article = 'Veganism'
granularity = 'daily'
start = '20150701'
end = '20190527'

endpoint = f'/metrics/pageviews/per-article/{project}/{access}/{agent}/{article}/{granularity}/{start}/{end}'
url = 'https://wikimedia.org/api/rest_v1' + endpoint
response = requests.get(url)
assert response.status_code == 200
df = pd.DataFrame(response.json()['items'])
df = df.rename(columns={'timestamp':'ds', 'views':'y'})
df['ds'] = pd.to_datetime(df['ds'], format='%Y%m%d%H')

```

- Nice use of datetime and I like the variable you set for the dataframe: 

```python

from fbprophet import Prophet
from fbprophet.plot import add_changepoints_to_plot

holidays = pd.DataFrame({
  'holiday': 'new_years',
  'ds': pd.to_datetime(['2015-01-01', '2016-01-01', '2017-01-01', '2018-01-01', 
                        '2019-01-01', '2020-01-01', '2021-01-01'], utc=True),
  'lower_window': -5,
  'upper_window': 5,
})

model = Prophet(daily_seasonality=False, holidays=holidays)
model.fit(df)
future = model.make_future_dataframe(periods=365*2)
forecast = model.predict(future)
fig1 = model.plot(forecast)
add_changepoints_to_plot(fig1.gca(), model, forecast)
fig2 = model.plot_components(forecast)

```

**Where you could of improved on:**

- Get into a habbit of setting a X and Y title (looks good): 


```python 

ax.set_xlabel("x label")


ax.set_ylabel("y label")

```

# Part 2 - Graphing 
***
**What went well:**

- Good job setting daily and high temp:

```python

ax = train.plot(x='ds', y='y', color='blue', label='Train', figsize=(16,4))
test.plot(x='ds', y='y', color='red', label='Test', ax=ax)
train.plot(x='ds', y='mean_baseline', color='black', 
           label='Mean Baseline, Train', ax=ax)
test.plot(x='ds', y='mean_baseline', color='black', 
          label='Mean Baseline, Test', ax=ax)
plt.title('Daily high temperature in Normal, IL');


```

- Good showing of daily high through matplotlib:

```python
ax = train.plot(x='ds', y='y', color='blue', label='Train', figsize=(16,4))
test.plot(x='ds', y='y', color='red', label='Test', ax=ax)
train.plot(x='ds', y='naive_baseline', color='black', 
           label='Naive Baseline, Train', ax=ax)
test.plot(x='ds', y='naive_baseline', color='black', 
          label='Naive Baseline, Test', ax=ax)
plt.title('Daily high temperature in Normal, IL');


```
- Great use of Prophet library and using the metrics .:

```python


ax = train.plot(x='ds', y='y', color='blue', label='train', figsize = (16, 4))
test.plot(x='ds', y='y', color='red', label = 'test', ax=ax)
train.plot(x='ds', y='mean_baseline', color='black', label='mean baseline, train', ax= ax)
test.plot(x='ds', y='mean_baseline', color='black', label='mean baseline, test', ax=ax)

ax = train.plot(x='ds', y='y', color='blue', label='train', figsize = (16, 4))
test.plot(x='ds', y='y', color='red', label = 'test', ax=ax)
train.plot(x='ds', y='naive_baseline', color='black', label='naive baseline, train', ax= ax)
test.plot(x='ds', y='naive_baseline', color='black', label='naive baseline, test', ax = ax);

ax = train.plot(x='ds', y='y', color='blue', label='train', figsize = (16, 4))
test.plot(x='ds', y='y', color='red', label = 'test', ax=ax)
train_forecast.plot(x='ds', y='yhat', color='black', label='prophet forecast, train',
                   ax=ax)
test_forecast.plot(x='ds', y='yhat', color='green', label='prophet forecast, test', ax=ax)
plt.title('daily high temp in Normal, IL')



from fbprophet.plot import plot_cross_validation_metric
fig = plot_cross_validation_metric(df_cv, metric='mae')




```

# What to work on
- [x] Comment more in code 
- [x] Add X and Y labels 
- [x] Come to PM for questions or help

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# Matthew Mastin- "Regression 1 Sprint Challenge"

# Part 1. Predict avocado sales
***
**What went well:**

-Good job imporing data and dataset:

```python



import pandas as pd
avocado_url = 'https://drive.google.com/uc?export=download&id=1ljE-LhCA9CCFvSIJYJ1ewX8JmdDpfuhi'
avocado = pd.read_csv(avocado_url)
assert avocado.shape == (169, 2)
import seaborn as sns




```

## 1.2. Plot the forecast

- Great job with the syntax here

```python



# adding prediction room

avocado = avocado.reindex(pd.date_range(start='2015-01-04', end='2019-03-25', freq='W'))
avocado.plot();



fig2 = m.plot_components(forecast)



```

## 1.3. Plot the forecast components


- Impressive work on finding Absolute Error and mean from sklearn:

```python

fig2 = m.plot_components(forecast)


```
## 1.4 Do time series cross-validation, with these parameters: horizon='90 days', period='45 days', initial='730 days'
**Where you could of improved on:**

-Easy syntax would be this way:


```python 

from fbprophet.diagnostics import cross_validation
df_cv = cross_validation(m, horizon='90 days', period='45 days', initial='730 days')


```

##1.5. Plot the cross-validation Mean Absolute Percentage Error: metric='mape' 
***
**What went well:**

- Clean and elegant: code. I'm impressed :) :

```python



%matplotlib inline
import seaborn as sns
sns.distplot(df['Incumbent Party Vote Share']);



```

#Stretch Goal (PARTIAL)

# What to work on
- [x] Comment more in code 
- [x] Less lines and more organized notebooks 
- [x] Come to PM for questions or help

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# Matthew Mastin- Regression 1 Sprint Challenge

# Part 1. Predict avocado sales 

***
**What went well:**

-Good collective use of imports and reading in data of the :

```python

import pandas as pd
avocado_url = 'https://drive.google.com/uc?export=download&id=1ljE-LhCA9CCFvSIJYJ1ewX8JmdDpfuhi'
avocado = pd.read_csv(avocado_url)
assert avocado.shape == (169, 2)
import seaborn as sns



```
#Part 2. Predict NYC apartment rent 

## Prophet Plotting 

- Good job with matplotlib and using this (industry standard) :

```python


# adding prediction room

avocado = avocado.reindex(pd.date_range(start='2015-01-04', end='2019-03-25', freq='W'))
avocado.plot();

# plotting predictions

fig1 = m.plot(forecast)

```

- Good use of [cross_validation](https://www.google.com) with Facebook's diagnostics API: 

```python

from fbprophet.diagnostics import cross_validation
avocado_cv = cross_validation(m, initial= '730 days',
                             horizon = '90 days', period = '45 days')
```

## Calculating MAPE(Mean Absolute Percentage Error)

- Impressive use of finding the MAPE!: 

```python
from fbprophet.plot import plot_cross_validation_metric
fig = plot_cross_validation_metric(avocado_cv, metric='mape')

```
# Part 2. Predict NYC apartment rent

## Setting Target

- Nice job setting feature and target of rent:  

```python
target = 'price'

rent['Mean Baseline'] = [rent[target].mean()] * len(rent)

```

## MAE

- Excellent work. I mean this because this is important when looking at features:

```python

from sklearn.metrics import mean_absolute_error, r2_score, mean_squared_error
print('MAE for mean baseline: ', mean_absolute_error(rent['price'], rent['Mean Baseline']))
print('R^2 for mean baseline: ', r2_score(rent['price'], rent['Mean Baseline']))
#MAE for mean baseline:  1055.7857771080928
#R^2 for mean baseline:  0.0

```
#What to work on
- [x] Comment more in code
- [x] Come to PM for questions or help
